t_sndrel(3)                                                 t_sndrel(3)

  tt__ssnnddrreell(())

  NNAAMMEE

    t_sndrel - initiate an orderly release

  SSYYNNOOPPSSIISS

    #include <xti.h>

    int t_sndrel(
        int fd)

  DDEESSCCRRIIPPTTIIOONN

    For transport providers of type T_COTS_ORD, the tt__ssnnddrreell(3) function is
    used to initiate an orderly release of the outgoing direction of data
    transfer and indicates to the transport provider that the transport user
    has no more data to send.

    PPaarraammeetteerrss     BBeeffoorree ccaallll     AAfftteerr ccaallll

    fd             x               /

    The argument fd identifies the local transport endpoint where the
    connection exists. After calling tt__ssnnddrreell(), the user may not send any
    more data over the connection. However, a user may continue to receive
    data if an orderly release indication has not been received. For transport
    providers of types other than T_COTS_ORD, this function fails with error
    [TNOTSUPPORT].

  VVAALLIIDD SSTTAATTEESS

    T_DATAXFER,T_INREL

  EERRRROORRSS

    On failure, t_errno is set to one of the following:

    [TBADF]
        The specified file descriptor does not refer to a transport endpoint.

    [TFLOW]
        O_NONBLOCK was set, but the flow control mechanism prevented the
        transport provider from accepting the function at this time.

    [TLOOK]
        An asynchronous event has occurred on this transport endpoint and
        requires immediate attention.

    [TNOTSUPPORT]
        This function is not supported by the underlying transport provider.

    [TOUTSTATE]
        The communications endpoint referenced by fd is not in one of the
        states in which a call to this function is valid.

    [TPROTO]
        This error indicates that a communication problem has been detected
        between XTI and the transport provider for which there is no other
        suitable XTI error (t_errno).

    [TSYSERR]
        A system error has occurred during execution of this function.

  RREETTUURRNN VVAALLUUEE

    Upon successful completion, a value of 0 is returned. Otherwise, a value
    of -1 is returned and t_errno is set to indicate an error.

  SSEEEE AALLSSOO

    _tt____gg_ee_tt_ii_nn_ff_oo(3)

    _tt____oo_pp_ee_nn(3)

    _tt____rr_cc_vv_rr_ee_ll(3)

  UUSSAAGGEE NNOOTTEESS

    The tt__ssnnddrreell function is not thread safe.

    The tt__ssnnddrreell function is not async-signal safe.

